MapboxDirections/MBDirectionOptions.swift

```diff
@@ -0,0 +1,449 @@
+     */
+    case full
+    
+    public init?(description: String) {
+        let granularity: RouteShapeResolution
+        switch description {
+        case "false":
+            granularity = .none
+        case "simplified":
+            granularity = .low
+        case "full":
+            granularity = .full
+        default:
+            return nil
+        }
+        self.init(rawValue: granularity.rawValue)
+    }
+    
+    public var description: String {
+        switch self {
+        case .none:
+            return "false"
+        case .low:
+            return "simplified"
+        case .full:
+            return "full"
+        }
+    }
+}
+
+/**
+ A system of units of measuring distances and other quantities.
+ */
+@objc(MBMeasurementSystem)
+public enum MeasurementSystem: UInt, CustomStringConvertible {
+    
+    /**
+     U.S. customary and British imperial units.
+     
+     Distances are measured in miles and feet.
+     */
+    case imperial
+    
+    /**
+     The metric system.
+     
+     Distances are measured in kilometers and meters.
+     */
+    case metric
+    
+    public init?(description: String) {
+        let measurementSystem: MeasurementSystem
+        switch description {
+        case "imperial":
+            measurementSystem = .imperial
+        case "metric":
+            measurementSystem = .metric
+        default:
+            return nil
+        }
+        self.init(rawValue: measurementSystem.rawValue)
+    }
+    
+    public var description: String {
+        switch self {
+        case .imperial:
+            return "imperial"
+        case .metric:
+            return "metric"
+        }
+    }
+}
+
+/**
+ A `RouteShapeFormat` indicates the format of a route’s shape in the raw HTTP response.
+ */
+@objc(MBInstructionFormat)
+public enum InstructionFormat: UInt, CustomStringConvertible {
+    /**
+     The route steps’ instructions are delivered in plain text format.
+     */
+    case text
+    /**
+     The route steps’ instructions are delivered in HTML format.
+     
+     Key phrases are boldfaced.
+     */
+    case html
+    
+    public init?(description: String) {
+        let format: InstructionFormat
+        switch description {
+        case "text":
+            format = .text
+        case "html":
+            format = .html
+        default:
+            return nil
+        }
+        self.init(rawValue: format.rawValue)
+    }
+    
+    public var description: String {
+        switch self {
+        case .text:
+            return "text"
+        case .html:
+            return "html"
+        }
+    }
+}
+
+@objc(MBDirecitonOptions)
+open class DirectionOptions: NSObject, NSSecureCoding, NSCopying {
```

Comment:
This is a stylistic nitpick, but the ordering of code for this class reads a little weird.

I generally try to keep things in the following order: 
`Class Constants` -> `Instance Properties` -> `Intializers` -> `Biz Logic` -> `Utility Methods` ->  `Delegate Conformance extensions` -> `Private Extensions`

Using a pattern like this tends to keep things consistent and enhances readability. It's just kind of weird when you're looking at an initializer and you encounter references to properties that you haven't seen yet.